packages/core/src/State.ts

```diff
@@ -39,7 +43,9 @@ export function isStateConfig<
 export const isState = isStateConfig;
 export class State<
   TContext extends MachineContext,
-  TEvent extends EventObject = EventObject,
+  TEvent extends EventObject,
+  _TActions extends ParameterizedObject,
+  TActor extends ProvidedActor,
```

Comment:
I wonder if this shouldn't receive `TChildren` already (preprocessed type). I'm still on the fence about this though